azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -934,7 +939,9 @@ public void cancelExecutingJobs(ExecutableFlow exFlow, String userId,
   @Override
   public String submitExecutableFlow(ExecutableFlow exflow, String userId)
     throws ExecutorManagerException {
-    synchronized (exflow) {
+
+    String exFlowKey = exflow.getProjectName() + "." + exflow.getId() + ".submitFlow";
+    synchronized (exFlowKey.intern()) {
```

Comment:
Is this change still needed until we have a better fix in place? This will only help with the race condition between scheduled flow and non-scheduled flow executions. 
The other change would fix the bigger vulnerability.